### PR TITLE
Add verify of keypair

### PR DESCRIPTION
**Pull request by sakridge**
#### Problem

After storing their keypair somewhere safe, at some point someone might want to verify that their keypair is still working and could sign a message. Or double check after they wrote it that they wrote it correctly and can remember the passphrase.

#### Summary of Changes

Add a 'verify' command to sign a message with a given keypair and then verify the signature.

Fixes #

### CODE REVIEW

**Comment by sakridge** [comment]
@t-nelson @jstarry this seemed like a good idea. Do we have something like it already?

**Comment by jstarry** [comment]
@sakridge seems like a good idea to me!

We don't have anything like this. A developer would have to compare the output of the `solana-keygen pubkey` command or diff the results of `solana-keygen recover`

File: keygen/src/keygen.rs [hunk 1]
```diff
@@ -353,6 +376,23 @@ fn main() -> Result<(), Box<dyn error::Error>> {
                 .collect::<Vec<_>>();
             thread::park();
         }
+        ("verify", Some(matches)) => {
+            let keypair = get_keypair_from_matches(matches)?;
+            let test_data = "test".as_bytes();
+            let signature = Signature::new(&keypair.sign(&test_data).to_bytes());
+            let pubkey = if matches.is_present("pubkey") {
```
Comment (t-nelson) [inline keygen/src/keygen.rs]: I think the `pubkey` should be required.  Using the one generated from the input keypair (assuming it resolves to a keypair) will always pass `signature.verify()` below.

File: keygen/src/keygen.rs [hunk 2]
```diff
@@ -353,6 +376,23 @@ fn main() -> Result<(), Box<dyn error::Error>> {
...
+                bs58::decode(matches.value_of("pubkey").unwrap())
+                    .into_vec()
+                    .unwrap()
+            } else {
+                keypair.pubkey().to_bytes().to_vec()
+            };
+            if signature.verify(&pubkey, &test_data) {
+                println!("good!");
```
Comment (t-nelson) [inline keygen/src/keygen.rs]: Could maybe spruce these up a bit 😉

Comment (sakridge) [reply keygen/src/keygen.rs]: yea.. any suggestions?

Comment (t-nelson) [reply keygen/src/keygen.rs]: How about `println!("Verification for public key - {}: Success", pubkey)` and likewise for failed?

File: keygen/src/keygen.rs [hunk 3]
```diff
@@ -58,6 +78,25 @@ fn main() -> Result<(), Box<dyn error::Error>> {
...
+                .arg(
+                    Arg::with_name("infile")
+                        .index(1)
+                        .value_name("PATH")
+                        .takes_value(true)
+                        .help("Path to keypair file"),
+                )
+                .arg(
+                    Arg::with_name("pubkey")
+                        .long("pubkey")
```
Comment (t-nelson) [inline keygen/src/keygen.rs]: `.index(2)`

File: keygen/src/keygen.rs [hunk 4]
```diff
@@ -58,6 +78,25 @@ fn main() -> Result<(), Box<dyn error::Error>> {
...
+                        .required(true)
+                        .value_name("PATH")
```
Comment (t-nelson) [inline keygen/src/keygen.rs]: `.value_name("BASE58_PUBKEY")`

**Pull request merged**